sdk/core/System.ClientModel/src/Pipeline/PipelinePolicy.cs

```diff
@@ -1,13 +1,46 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System.Collections.Generic;
 using System.Threading.Tasks;
 
 namespace System.ClientModel.Primitives;
 
 public abstract class PipelinePolicy
 {
-    public abstract void Process(PipelineMessage message, PipelineProcessor pipeline);
+    internal int PipelineIndex { get; set; }
```

Comment:
This will not work since a policy could be included in multiple pipelines with different indexes.  The policy's own index should be on the stack in the ProcessNext method.

Reply:
Maybe we can make it an internal property on message instead.

Reply:
Addressed by passing currentIndex as part of the public API